src/tiertwo/init.cpp

```diff
@@ -0,0 +1,153 @@
+{
+    RegisterValidationInterface(&g_budgetman);
+    RegisterValidationInterface(&masternodePayments);
+    if (activeMasternodeManager) RegisterValidationInterface(activeMasternodeManager);
+}
+
+void DumpTierTwo()
+{
+    DumpMasternodes();
+    DumpBudgets(g_budgetman);
+    DumpMasternodePayments();
+}
+
+void SetBudgetFinMode(const std::string& mode)
+{
+    g_budgetman.strBudgetMode = mode;
+    LogPrintf("Budget Mode %s\n", g_budgetman.strBudgetMode);
+}
+
+bool InitActiveMN()
+{
+    fMasterNode = gArgs.GetBoolArg("-masternode", DEFAULT_MASTERNODE);
+    if ((fMasterNode || masternodeConfig.getCount() > -1) && fTxIndex == false) {
+        return UIError(strprintf(_("Enabling Masternode support requires turning on transaction indexing."
+                                   "Please add %s to your configuration and start with %s"), "txindex=1", "-reindex"));
+    }
+
+    if (fMasterNode) {
+        const std::string& mnoperatorkeyStr = gArgs.GetArg("-mnoperatorprivatekey", "");
+        const bool fDeterministic = !mnoperatorkeyStr.empty();
+        LogPrintf("IS %s MASTERNODE\n", (fDeterministic ? "DETERMINISTIC " : ""));
+
+        if (fDeterministic) {
+            // Check enforcement
+            if (!deterministicMNManager->IsDIP3Enforced()) {
+                const std::string strError = strprintf(
+                        _("Cannot start deterministic masternode before enforcement. Remove %s to start as legacy masternode"),
+                        "-mnoperatorprivatekey");
+                LogPrintf("-- ERROR: %s\n", strError);
+                return UIError(strError);
+            }
+            // Create and register activeMasternodeManager
+            activeMasternodeManager = new CActiveDeterministicMasternodeManager();
+            auto res = activeMasternodeManager->SetOperatorKey(mnoperatorkeyStr);
+            if (!res) { return UIError(res.getError()); }
+            // Init active masternode
+            const CBlockIndex* pindexTip = WITH_LOCK(cs_main, return chainActive.Tip(););
+            activeMasternodeManager->Init(pindexTip);
+        } else {
+            // Check enforcement
+            if (deterministicMNManager->LegacyMNObsolete()) {
+                const std::string strError = strprintf(
+                        _("Legacy masternode system disabled. Use %s to start as deterministic masternode"),
+                        "-mnoperatorprivatekey");
+                LogPrintf("-- ERROR: %s\n", strError);
+                return UIError(strError);
+            }
+            auto res = initMasternode(gArgs.GetArg("-masternodeprivkey", ""), gArgs.GetArg("-masternodeaddr", ""),
+                                      true);
+            if (!res) { return UIError(res.getError()); }
+        }
+    }
+    // All good
+    return true;
+}
+
+void StartTierTwoThreadsAndScheduleJobs(boost::thread_group& threadGroup, CScheduler& scheduler)
+{
+    threadGroup.create_thread(std::bind(&ThreadCheckMasternodes));
+}
```

Comment:
since you are going to rebase this due to conflict, might as well fix this missing blank line at the end of `tiertwo/init.cpp`.

Reply:
jump line added.